File: src/game.cpp

// ゲーム内の入力に変換するためのクラス
#include "main.h"
#include "game.h"

bool RB,LB,EB,AT,Key,rB,lB,eB;
int barx; 

GAME::GAME()
{
	main = new MAIN;
}


GAME::~GAME()
{
	delete main;
}

bool GAME::GameStart(){
	if(main->All() == 0)
		return false;
	return true;
}	

void GAME::RButton(){ // 左入力
	RB = true;
	Key = true;
}
void GAME::LButton(){ // 右入力
	LB = true;
	Key = true;
}
void GAME::BarMove( int vx ){ // バーのx座標更新
	barx = vx;
}
void GAME::EnterButton(){ // 決定ボタン
	EB = true;
	Key = true;
}
void GAME::Auto( bool on){
	AT = on;
}
void GAME::KeyEnd(){
	Key = false;
}

File: libBlockCrashBridge/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DxLibDLL;

namespace libBlockCrashBridge
{
    public class Game
    {
        private int vx;
        private int endflag;
        private Main main;
        private Input input;

        public Game()
        {
            main = new Main(input);
        }

        public bool GameStart()
        {
            return main.All();
        }

        public void RButton()
        {
            input.RB = true;
            input.Key = true;
        }

        public void LButton()
        {
            input.LB = true;
            input.Key = true;
        }

        public void BarMove(int vx)
        {
            input.barx = vx;
        }

        public void EnterButton()
        {
            input.EB = true;
            input.Key = true;
        }

        public void Auto(bool on)
        {
            input.AT = on;
        }

        public void KeyEnd()
        {
            input.Key = false;
        }
    }
}
